pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -175,9 +176,35 @@ func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcil
 	}
 	// provisionedPhase is the status of the machine when it is in the `Provisioned` state
 	provisionedPhase := "Provisioned"
-	if machine.Status.Phase == nil || *machine.Status.Phase != provisionedPhase {
-		// Phase can be nil and should be ignored by WMCO
-		// If the Machine is not in provisioned state, WMCO shouldn't care about it
+	// readyPhase is the status of the machine when it is in the `Running` state, indicating that it is configured into a node
+	runningPhase := "Running"
+	if machine.Status.Phase == nil {
+		// Phase is nil and should be ignored by WMCO until phase is set
+		return reconcile.Result{}, nil
+	} else if *machine.Status.Phase == runningPhase {
+		// Machine has been configured into a node, we need to ensure that the version annotation exists. If it doesn't
+		// the machine was not fully configured and needs to be configured properly.
+		if machine.Status.NodeRef == nil {
+			// NodeRef missing. Requeue and hope it is created. It never being created indicates an issue with the
```

Comment:
Or do we delete the machine? But we could be running into a race condition here.